subsys/nrf_security/src/drivers/Kconfig

```diff
@@ -4,56 +4,230 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_SUPPORT
+# Legacy config that needs to be removed later. The Oberon PSA driver is now the fallback option
+# for all the crypto operations when hardware acceleration is not available.
+config PSA_CRYPTO_DRIVER_OBERON
+	# The Oberon driver is required to provide HKDF needed for protected storage and bultin keys
+	prompt "Oberon PSA driver" if !(TFM_PARTITION_PROTECTED_STORAGE || TFM_CRYPTO_BUILTIN_KEYS)
 	bool
 	default y
-	depends on MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_AES || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_CHACHA20
+	help
+	  This configuration enables the usage of the Oberon PSA driver. Note that the Oberon
+	  PSA core is always included in the build.
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR
+config PSA_CRYPTO_DRIVER_CC3XX
+	prompt "CryptoCell PSA driver"
+	bool
+	help
+	  This configuration enables the usage of CryptoCell for the supported operations.
+	  Disabling this option will result in all crypto operations being handled by
+	  the Oberon PSA driver. However, please note that disabling this option does not
+	  affect the DRBG algorithms; even when disabled, DRBG will still be provided by CryptoCell.
+	  The CryptoCell PSA driver does not support key size configurations. When enabled,
+	  all supported key sizes are included in the build.
+
+menu "Choose DRBG algorithm"
+config PSA_WANT_ALG_CTR_DRBG
+	prompt "Enable CTR_DRBG"
+	bool
+	default y if !PSA_WANT_ALG_HMAC_DRBG
+	depends on PSA_WANT_GENERATE_RANDOM
+
+config PSA_WANT_ALG_HMAC_DRBG
+	prompt "Enable HMAC_DRBG"
+	bool
+	depends on PSA_WANT_GENERATE_RANDOM
+
+# The Oberon PSA core requires the USE symbols to choose DRBG algorithm.
+config PSA_USE_CTR_DRBG_DRIVER
+	bool
+	default y if PSA_WANT_ALG_CTR_DRBG
+
+config PSA_USE_HMAC_DRBG_DRIVER
+	bool
+	default y if PSA_WANT_ALG_HMAC_DRBG
+
+endmenu
+
+menu "CryptoCell PSA Driver Configuration"
+if PSA_CRYPTO_DRIVER_CC3XX
+
+config PSA_USE_CC3XX_CIPHER_DRIVER
+	prompt "Enable CryptoCell for the PSA cipher APIs"
 	bool
 	default y
-	depends on PSA_WANT_KEY_TYPE_ECC_KEY_PAIR && \
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_CTR            || \
```

Comment:
According to the Oberon docs this option should not depend on these ALG's.

> The meaning of a PSA_USE_XYZ directive is as follows: "Assuming that the application wants to use at least one algorithm implemented in this driver, link the relevant code parts of this driver to the final binary image - otherwise don't link this driver's code".

I don't see the need to diverge from Oberon's system here, although I could be missing something.

We would need "depends on" if the USE symbol was defined like this:

Link the relevant code parts of this driver to the final binary image - otherwise don't link this driver's code.

The NEED symbol should check for WANT. Checking WANT in the USE symbol as well is superflous for CIPHER because CC3XX implements all the same cipher algorithms that the Oberon PSA Core does.

Reply:
I would not call it a deviation of their system to be honest. I see your point though and I agree that I added some more logic to their logic.

So the difference if I remove this is that a user will see all the USE Kconfigs enabled even if they don't want any of the algorithms that they provide.  I thought that doing it this way improves usability a bit since the users will only see automatically selected the drivers that they will get. 

This is not a big change and I am OK to remove all the depends if you think that it is better to be closer to what they say in the description.

This should not affect functionality at all I think so I am OK to update if you want.

Reply:
I do wish that we update.

I don't feel that this added complexity is worth the benefits that you describe.
It is more difficult to understand the system if we diverge without documenting our divergence.

It is also unfortunate if we give the impression to the user that the USE symbol determines what is used in the build.

If they are all enabled it will be more clear to the user that this is not what these symbols are for.

If you wish to make it less confusing we could request a renaming from Oberon.

Reply:
Unnecessary dependencies can also create dependency loops, although I don't know if we will have this problem here.

Reply:
I don't have a strong opinion on this, I will remove the depends.